packages/PHPStanStaticTypeMapper/TypeMapper/ClosureTypeMapper.php

```diff
@@ -30,7 +34,23 @@ public function getNodeClass(): string
      */
     public function mapToPHPStanPhpDocTypeNode(Type $type): TypeNode
     {
-        return $type->toPhpDocNode();
+        $typeNode = $type->toPhpDocNode();
+
+        $phpDocNodeTraverser = new PhpDocNodeTraverser();
+        $phpDocNodeTraverser->traverseWithCallable($typeNode, '', function (AstNode $astNode) {
+            if (! $astNode instanceof IdentifierTypeNode) {
+                return null;
+            }
+
+            if ($astNode->name !== 'Closure') {
+                return null;
+            }
+
+            $astNode = new FullyQualifiedIdentifierTypeNode('Closure');
+            return $astNode;
+        });
+
+        return $typeNode;
```

Comment:
@TomasVotruba here the patch, replace `IdentifierTypeNode` on transformed `PhpDocNode` to `FullyQualifiedIdentifierTypeNode`